helix-core/src/main/java/org/apache/helix/task/RuntimeJobDag.java

```diff
@@ -198,11 +198,15 @@ public void generateJobList() {
     resetJobListAndDependencyMaps();
     computeIndependentNodes();
     _readyJobList.addAll(_independentNodes);
-    if (_isJobQueue && _readyJobList.size() > 0) {
+    if (_isJobQueue && !_readyJobList.isEmpty()) {
       // For job queue, only get number of parallel jobs to run in the ready list.
       for (int i = 1; i < _numParallelJobs; i++) {
-        if (_parentsToChildren.containsKey(_readyJobList.peekLast())) {
-          _readyJobList.offer(_parentsToChildren.get(_readyJobList.peekLast()).iterator().next());
+        String parent = _readyJobList.peekLast();
+        Set<String> children = _parentsToChildren.get(parent);
+        if (children != null) {
+          _readyJobList.offer(children.iterator().next());
+        } else {
+          break;
```

Comment:
NIT: it would be better make it as:

if (children == null) {
break;
} 

_readyJobList.offer(children.iterator().next());

Reply:
Sure I can change this if you like.

Reply:
Done